service/src/test/java/bio/terra/workspace/common/utils/WorkspaceActivityLogHookTest.java

```diff
@@ -0,0 +1,161 @@
+
+import bio.terra.stairway.FlightDebugInfo;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.db.WorkspaceActivityLogDao;
+import bio.terra.workspace.db.WorkspaceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.job.JobService;
+import bio.terra.workspace.service.job.JobServiceTestFlight;
+import bio.terra.workspace.service.workspace.model.OperationType;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+public class WorkspaceActivityLogHookTest extends BaseUnitTest {
+  private static final AuthenticatedUserRequest testUser =
+      new AuthenticatedUserRequest()
+          .subjectId("StairwayUnit")
+          .email("stairway@unit.com")
+          .token(Optional.of("not-a-real-token"));
+
+  @Autowired private JobService jobService;
+  @Autowired private WorkspaceDao workspaceDao;
+  @Autowired private WorkspaceActivityLogDao activityLogDao;
+  @MockBean private SamService mockSamService;
+
+  @BeforeEach
+  @SuppressFBWarnings(value = "DE_MIGHT_IGNORE", justification = "Mockito flakiness")
+  void setup() {
+    try {
+      Mockito.doReturn(true).when(mockSamService.isAuthorized(any(), any(), any(), any()));
+    } catch (Exception e) {
+      // How does a mock even throw an exception?
+    }
+  }
+
+  /**
+   * Reset the {@link JobService} {@link FlightDebugInfo} after each test so that future submissions
+   * aren't affected.
+   */
+  @AfterEach
+  void clearFlightDebugInfo() {
+    jobService.setFlightDebugInfoForTest(null);
+  }
+
+  @Test
+  void createFlightSucceeds_activityLogUpdated() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var emptyChangedDate = activityLogDao.getLastUpdatedDate(workspaceUuid);
+    assertTrue(emptyChangedDate.isEmpty());
+
+    runFlight("a creation flight", workspaceUuid, OperationType.CREATE);
+    var changedDate = activityLogDao.getLastUpdatedDate(workspaceUuid);
+    assertTrue(changedDate.isPresent());
+  }
+
+  @Test
+  void deleteFlightSucceeds_activityLogUpdated() {
+    UUID workspaceUuid = UUID.randomUUID();
+    var emptyChangedDate = activityLogDao.getLastUpdatedDate(workspaceUuid);
+    assertTrue(emptyChangedDate.isEmpty());
+
+    runFlight("a creation flight", workspaceUuid, OperationType.DELETE);
+
+    var changedDate = activityLogDao.getLastUpdatedDate(workspaceUuid);
+    assertTrue(changedDate.isPresent());
+  }
+
+  @Test
+  void unknownFlightSucceeds_activityLogNotUpdated() {
```

Comment:
I thought we throw for UNKNOWN?

flightWithUnknownOperationType_throws

opt Dao is the place that throws for UNKNOWN operationType. Test for "throw for UNKNOWN" only needs to be in Dao Test, it doesn't need to be in this test also.

Reply:
this test uses jobService.waitForJob which doesn't re-throw flight. I made operationtype required not though so this will throw early at jobbuilder.